hi-ml-azure/tests/testazure/test_azure_util.py

```diff
@@ -1006,3 +1008,81 @@ def test_is_local_rank_zero() -> None:
 
     with mock.patch.dict(os.environ, {util.ENV_GLOBAL_RANK: "1", util.ENV_LOCAL_RANK: "1"}):
         assert not util.is_local_rank_zero()
+
+
+@pytest.mark.parametrize("overwrite", [True, False])
+@pytest.mark.parametrize("show_progress", [True, False])
+def test_download_from_datastore(tmp_path: Path, overwrite: bool, show_progress: bool) -> None:
+    """
+    Test that download_from_datastore successfully downloads file from Blob Storage.
+    Note that this will temporarily upload a file to the default datastore of the default workspace -
+    (determined by either a config.json file, or by specifying workspace settings in the environment variables).
+    After the test has completed, the blob will be deleted.
+    """
+    ws = DEFAULT_WORKSPACE.workspace
+    default_datastore: AzureBlobDatastore = ws.get_default_datastore()
+    dummy_file_content = "Hello world"
+
+    # Create a dummy data file and upload to datastore
+    dummy_filename = "dummy_data.txt"
+    data_to_upload_path = tmp_path / dummy_filename
+    data_to_upload_path.touch()
+    data_to_upload_path.write_text(dummy_file_content)
+    test_data_path_remote = "test_data"
+    default_datastore.upload(str(tmp_path), test_data_path_remote, overwrite=False)
+
+    # Check that the file doesn't currently exist at download location
+    downloaded_data_path = tmp_path / "downloads"
+    assert not downloaded_data_path.exists()
+
+    # Now attempt to download
+    util.download_from_datastore(default_datastore.name, test_data_path_remote, downloaded_data_path,
+                                 aml_workspace=ws, overwrite=overwrite, show_progress=show_progress)
+    assert downloaded_data_path.exists()
+
+    # Delete the file from Blob Storage
+    expected_remote_path = Path(test_data_path_remote) / dummy_filename
+    container = default_datastore.container_name
+    existing_blobs = list(default_datastore.blob_service.list_blobs(prefix=str(expected_remote_path.as_posix()),
+                                                                    container_name=container))
```

Comment:
maybe check that files exist in blob storage right after upload? Otherwise, errors in download can either mean a bug there, or a bug in upload?